tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,771 @@
+				Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+				By("Creating external FRR pod on master node")
+				frrPod := deployTestPods(addressPool, hubIPaddresses, externalAdvertisedIPv4Routes,
+					externalAdvertisedIPv6Routes)
+
+				By("Creating BGP Peers")
+				createBGPPeerAndVerifyIfItsReady(ipv4metalLbIPList[0], "", 64500,
+					false, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, "IPv4", removePrefixFromIPList(nodeAddrList), addressPool, 32)
+
+				By("Create a frrconfiguration allow all")
+				err = createFrrConfiguration("frrconfig-allow-all", ipv4metalLbIPList[0],
+					64500, nil, false)
+				Expect(err).ToNot(HaveOccurred(), "Fail to create a FRRconfiguration with allow all")
+
+				frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+					LabelSelector: "app=frr-k8s",
+				})
+				Expect(err).ToNot(HaveOccurred(), "Fail to find Frrk8 pod list")
+
+				By("Verify that the node FRR pods advertises two routes")
+				err = verifyExternalAdvertisedRoutes(frrPod, ipv4NodeAddrList, externalAdvertisedIPv4Routes)
+				Expect(err).ToNot(HaveOccurred(), "Fail to find advertised routes")
+
+				By("Validate BGP received routes")
+				Eventually(func() string {
+					// Get the routes
+					routes, err := frr.VerifyBGPReceivedRoutesOnFrrNodes(frrk8sPods)
+					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
+
+					// Return the routes to be checked
+					return routes
+				}, 60*time.Second, 5*time.Second).Should(ContainSubstring(externalAdvertisedIPv4Routes[1]),
+					"Fail to find received routes")
+			})
+
+		It("Verify the FRR node only receives routes that are configured in the allowed prefixes",
+			reportxml.ID("74272"), func() {
+
+				By("Creating a new instance of MetalLB Speakers on workers")
+				err = metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+				Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+				frrPod := deployTestPods(addressPool, hubIPaddresses, externalAdvertisedIPv4Routes,
+					externalAdvertisedIPv6Routes)
+
+				By("Creating BGP Peers")
+				createBGPPeerAndVerifyIfItsReady(ipv4metalLbIPList[0], "", 64500, false, frrk8sPods)
+
+				By("Checking that BGP session is established and up")
+				verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+				By("Validating BGP route prefix")
+				validatePrefix(frrPod, "IPv4", removePrefixFromIPList(nodeAddrList), addressPool, 32)
+
+				By("Create a frrconfiguration with prefix filter")
+				err := createFrrConfiguration("frrconfig-filtered", ipv4metalLbIPList[0],
+					64500, []string{externalAdvertisedIPv4Routes[0], externalAdvertisedIPv6Routes[0]}, false)
```

Comment:
same, declare a filteredIPs var to make this more readable and reuse later.
also, the IPv6 prefix is not verified later, is that on purpose?

Reply:
Added.
Currently we are having issues deploying IPv6 in our lab. We will add the IPv6 when it is possible to test.